File: octgnFX/agsxmpp/trunk/samples/VS2005/MiniClientVB/frmVcard.vb
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Imports agsXMPP
Imports agsXMPP.protocol
Imports agsXMPP.protocol.client
Imports agsXMPP.protocol.iq.vcard

Namespace MiniClient

	Public Class frmVcard
	Inherits System.Windows.Forms.Form
		Private components As System.ComponentModel.Container = Nothing
		Private _connection As XmppClientConnection
		Private packetId As String
		Private label3 As System.Windows.Forms.Label
		Private txtBirthday As System.Windows.Forms.TextBox
		Private label1 As System.Windows.Forms.Label
		Private label5 As System.Windows.Forms.Label
		Private txtFullname As System.Windows.Forms.TextBox
		Private label2 As System.Windows.Forms.Label
		Private picPhoto As System.Windows.Forms.PictureBox
		Private txtDescription As System.Windows.Forms.TextBox
		Private label4 As System.Windows.Forms.Label
		Private cmdClose As System.Windows.Forms.Button
		Private txtNickname As System.Windows.Forms.TextBox
		
		Public Sub New(ByVal jid As Jid, ByVal con As XmppClientConnection)
			InitializeComponent
			_connection = con
			Me.Text = "Vcard from: " + jid.Bare
			Dim viq As VcardIq = New VcardIq(IqType.get, New Jid(jid.Bare))
			packetId = viq.Id
			con.IqGrabber.SendIq(viq, AddressOf VcardResult, Nothing)
		End Sub

		Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not (components Is Nothing) Then
					components.Dispose
				End If
			End If
			MyBase.Dispose(disposing)
			If Not (_connection.IqGrabber Is Nothing) Then
				_connection.IqGrabber.Remove(packetId)
			End If
		End Sub

		Private Sub InitializeComponent()
			Me.txtNickname = New System.Windows.Forms.TextBox
			Me.cmdClose = New System.Windows.Forms.Button
			Me.label4 = New System.Windows.Forms.Label
			Me.txtDescription = New System.Windows.Forms.TextBox
			Me.picPhoto = New System.Windows.Forms.PictureBox
			Me.label2 = New System.Windows.Forms.Label
			Me.txtFullname = New System.Windows.Forms.TextBox
			Me.label5 = New System.Windows.Forms.Label
			Me.label1 = New System.Windows.Forms.Label
			Me.txtBirthday = New System.Windows.Forms.TextBox
			Me.label3 = New System.Windows.Forms.Label
			Me.SuspendLayout
			'
			'txtNickname
			'
			Me.txtNickname.Location = New System.Drawing.Point(128, 48)
			Me.txtNickname.Name = "txtNickname"
			Me.txtNickname.ReadOnly = true
			Me.txtNickname.Size = New System.Drawing.Size(176, 20)
			Me.txtNickname.TabIndex = 1
			Me.txtNickname.Text = ""
			'
			'cmdClose
			'
			Me.cmdClose.FlatStyle = System.Windows.Forms.FlatStyle.System
			Me.cmdClose.Location = New System.Drawing.Point(120, 304)
			Me.cmdClose.Name = "cmdClose"
			Me.cmdClose.Size = New System.Drawing.Size(96, 24)
			Me.cmdClose.TabIndex = 4
			Me.cmdClose.Text = "Close"
			AddHandler Me.cmdClose.Click, AddressOf Me.cmdClose_Click
			'
			'label4
			'
			Me.label4.Location = New System.Drawing.Point(16, 48)
			Me.label4.Name = "label4"
			Me.label4.Size = New System.Drawing.Size(96, 16)
			Me.label4.TabIndex = 3
			Me.label4.Text = "Nick Name:"
			'
			'txtDescription
			'
			Me.txtDescription.Location = New System.Drawing.Point(128, 112)
			Me.txtDescription.Multiline = true
			Me.txtDescription.Name = "txtDescription"
			Me.txtDescription.ReadOnly = true
			Me.txtDescription.ScrollBars = System.Windows.Forms.ScrollBars.Vertical
			Me.txtDescription.Size = New System.Drawing.Size(176, 64)
			Me.txtDescription.TabIndex = 3
			Me.txtDescription.Text = ""
			'
			'picPhoto
			'
			Me.picPhoto.Location = New System.Drawing.Point(128, 192)
			Me.picPhoto.Name = "picPhoto"
			Me.picPhoto.Size = New System.Drawing.Size(176, 104)
			Me.picPhoto.TabIndex = 9
			Me.picPhoto.TabStop = false
			'
			'label2
			'
			Me.label2.Location = New System.Drawing.Point(16, 88)
			Me.label2.Name = "label2"
			Me.label2.Size = New System.Drawing.Size(104, 16)
			Me.label2.TabIndex = 1
			Me.label2.Text = "Birthday:"
			'
			'txtFullname
			'
			Me.txtFullname.Location = New System.Drawing.Point(128, 16)
			Me.txtFullname.Name = "txtFullname"
			Me.txtFullname.ReadOnly = true
			Me.txtFullname.Size = New System.Drawing.Size(176, 20)
			Me.txtFullname.TabIndex = 0
			Me.txtFullname.Text = ""
			'
			'label5
			'
			Me.label5.Location = New System.Drawing.Point(16, 192)
			Me.label5.Name = "label5"
			Me.label5.Size = New System.Drawing.Size(104, 16)
			Me.label5.TabIndex = 8
			Me.label5.Text = "Photo:"
			'
			'label1
			'
			Me.label1.Location = New System.Drawing.Point(16, 16)
			Me.label1.Name = "label1"
			Me.label1.Size = New System.Drawing.Size(96, 24)
			Me.label1.TabIndex = 0
			Me.label1.Text = "Full Name:"
			'
			'txtBirthday
			'
			Me.txtBirthday.Location = New System.Drawing.Point(128, 80)
			Me.txtBirthday.Name = "txtBirthday"
			Me.txtBirthday.ReadOnly = true
			Me.txtBirthday.Size = New System.Drawing.Size(176, 20)
			Me.txtBirthday.TabIndex = 2
			Me.txtBirthday.Text = ""
			'
			'label3
			'
			Me.label3.Location = New System.Drawing.Point(16, 120)
			Me.label3.Name = "label3"
			Me.label3.Size = New System.Drawing.Size(104, 16)
			Me.label3.TabIndex = 2
			Me.label3.Text = "Description:"
			'
			'frmVcard
			'
			Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
			Me.ClientSize = New System.Drawing.Size(328, 336)
			Me.Controls.Add(Me.cmdClose)
			Me.Controls.Add(Me.picPhoto)
			Me.Controls.Add(Me.label5)
			Me.Controls.Add(Me.txtNickname)
			Me.Controls.Add(Me.txtDescription)
			Me.Controls.Add(Me.txtBirthday)
			Me.Controls.Add(Me.txtFullname)
			Me.Controls.Add(Me.label4)
			Me.Controls.Add(Me.label3)
			Me.Controls.Add(Me.label2)
			Me.Controls.Add(Me.label1)
			Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle
			Me.MaximizeBox = false
			Me.Name = "frmVcard"
			Me.Text = "frmVcard"
			Me.ResumeLayout(false)
		End Sub

        Private Sub VcardResult(ByVal sender As Object, ByVal iq As agsXMPP.protocol.client.IQ, ByVal data As Object)
            If InvokeRequired Then
                BeginInvoke(CType(AddressOf VcardResult, agsXMPP.IqCB), New Object() {sender, iq, data})
                Return
            End If

            If iq.Type = IqType.result Then
                Dim vcard As Vcard = iq.Vcard
                If Not (vcard Is Nothing) Then
                    txtFullname.Text = vcard.Fullname
                    txtNickname.Text = vcard.Nickname
                    txtBirthday.Text = vcard.Birthday.ToString
                    txtDescription.Text = vcard.Description
                    Dim photo As Photo = vcard.Photo
                    If Not (photo Is Nothing) Then
                        picPhoto.Image = vcard.Photo.Image
                    End If
                End If
            End If
        End Sub

		Private Sub cmdClose_Click(ByVal sender As Object, ByVal e As System.EventArgs)
			Me.Close
		End Sub
	End Class 
End Namespace

File: octgnFX/agsxmpp/trunk/samples/VS2005/MiniClient/frmVcard.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

using agsXMPP;
using agsXMPP.protocol;
using agsXMPP.protocol.client;
using agsXMPP.protocol.iq.vcard;

namespace MiniClient
{
	/// <summary>
	/// Zusammenfassung für frmVcard.
	/// </summary>
	public class frmVcard : System.Windows.Forms.Form
	{
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.Label label3;
		private System.Windows.Forms.Label label4;
		/// <summary>
		/// Erforderliche Designervariable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		private XmppClientConnection _connection;
		private System.Windows.Forms.TextBox txtFullname;
		private System.Windows.Forms.TextBox txtNickname;
		private System.Windows.Forms.TextBox txtBirthday;
		private System.Windows.Forms.Label label5;
		private System.Windows.Forms.PictureBox picPhoto;
		private System.Windows.Forms.Button cmdClose;
		private System.Windows.Forms.TextBox txtDescription;
		string	packetId;

		public frmVcard(Jid jid, XmppClientConnection con)
		{
			
			InitializeComponent();
			
			_connection = con;

			this.Text = "Vcard from: " + jid.Bare;

			VcardIq viq = new VcardIq(IqType.get, new Jid(jid.Bare));
			packetId = viq.Id;
			con.IqGrabber.SendIq(viq, new IqCB(VcardResult), null);			
		}

		/// <summary>
		/// Die verwendeten Ressourcen bereinigen.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
			// Remove the packet from the Tracker in case we Close the from before getting
			// the result. The callback yould cause a nullpointer exception
			if (_connection.IqGrabber != null)
				_connection.IqGrabber.Remove(packetId);
		}

		#region Vom Windows Form-Designer generierter Code
		/// <summary>
		/// Erforderliche Methode für die Designerunterstützung. 
		/// Der Inhalt der Methode darf nicht mit dem Code-Editor geändert werden.
		/// </summary>
		private void InitializeComponent()
		{
			this.label1 = new System.Windows.Forms.Label();
			this.label2 = new System.Windows.Forms.Label();
			this.label3 = new System.Windows.Forms.Label();
			this.label4 = new System.Windows.Forms.Label();
			this.txtFullname = new System.Windows.Forms.TextBox();
			this.txtBirthday = new System.Windows.Forms.TextBox();
			this.txtDescription = new System.Windows.Forms.TextBox();
			this.txtNickname = new System.Windows.Forms.TextBox();
			this.label5 = new System.Windows.Forms.Label();
			this.picPhoto = new System.Windows.Forms.PictureBox();
			this.cmdClose = new System.Windows.Forms.Button();
			this.SuspendLayout();
			// 
			// label1
			// 
			this.label1.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.label1.Location = new System.Drawing.Point(16, 16);
			this.label1.Name = "label1";
			this.label1.Size = new System.Drawing.Size(96, 24);
			this.label1.TabIndex = 0;
			this.label1.Text = "Full Name:";
			// 
			// label2
			// 
			this.label2.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.label2.Location = new System.Drawing.Point(16, 88);
			this.label2.Name = "label2";
			this.label2.Size = new System.Drawing.Size(104, 16);
			this.label2.TabIndex = 1;
			this.label2.Text = "Birthday:";
			// 
			// label3
			// 
			this.label3.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.label3.Location = new System.Drawing.Point(16, 120);
			this.label3.Name = "label3";
			this.label3.Size = new System.Drawing.Size(104, 16);
			this.label3.TabIndex = 2;
			this.label3.Text = "Description:";
			// 
			// label4
			// 
			this.label4.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.label4.Location = new System.Drawing.Point(16, 48);
			this.label4.Name = "label4";
			this.label4.Size = new System.Drawing.Size(96, 16);
			this.label4.TabIndex = 3;
			this.label4.Text = "Nick Name:";
			// 
			// txtFullname
			// 
			this.txtFullname.Location = new System.Drawing.Point(128, 16);
			this.txtFullname.Name = "txtFullname";
			this.txtFullname.ReadOnly = true;
			this.txtFullname.Size = new System.Drawing.Size(176, 20);
			this.txtFullname.TabIndex = 0;
			this.txtFullname.Text = "";
			// 
			// txtBirthday
			// 
			this.txtBirthday.Location = new System.Drawing.Point(128, 80);
			this.txtBirthday.Name = "txtBirthday";
			this.txtBirthday.ReadOnly = true;
			this.txtBirthday.Size = new System.Drawing.Size(176, 20);
			this.txtBirthday.TabIndex = 2;
			this.txtBirthday.Text = "";
			// 
			// txtDescription
			// 
			this.txtDescription.Location = new System.Drawing.Point(128, 112);
			this.txtDescription.Multiline = true;
			this.txtDescription.Name = "txtDescription";
			this.txtDescription.ReadOnly = true;
			this.txtDescription.ScrollBars = System.Windows.Forms.ScrollBars.Vertical;
			this.txtDescription.Size = new System.Drawing.Size(176, 64);
			this.txtDescription.TabIndex = 3;
			this.txtDescription.Text = "";
			// 
			// txtNickname
			// 
			this.txtNickname.Location = new System.Drawing.Point(128, 48);
			this.txtNickname.Name = "txtNickname";
			this.txtNickname.ReadOnly = true;
			this.txtNickname.Size = new System.Drawing.Size(176, 20);
			this.txtNickname.TabIndex = 1;
			this.txtNickname.Text = "";
			// 
			// label5
			// 
			this.label5.Font = new System.Drawing.Font("Microsoft Sans Serif", 8.25F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((System.Byte)(0)));
			this.label5.Location = new System.Drawing.Point(16, 192);
			this.label5.Name = "label5";
			this.label5.Size = new System.Drawing.Size(104, 16);
			this.label5.TabIndex = 8;
			this.label5.Text = "Photo:";
			// 
			// picPhoto
			// 
			this.picPhoto.Location = new System.Drawing.Point(128, 192);
			this.picPhoto.Name = "picPhoto";
			this.picPhoto.Size = new System.Drawing.Size(176, 104);
			this.picPhoto.TabIndex = 9;
			this.picPhoto.TabStop = false;
			// 
			// cmdClose
			// 
			this.cmdClose.FlatStyle = System.Windows.Forms.FlatStyle.System;
			this.cmdClose.Location = new System.Drawing.Point(120, 304);
			this.cmdClose.Name = "cmdClose";
			this.cmdClose.Size = new System.Drawing.Size(96, 24);
			this.cmdClose.TabIndex = 4;
			this.cmdClose.Text = "Close";
			this.cmdClose.Click += new System.EventHandler(this.cmdClose_Click);
			// 
			// frmVcard
			// 
			this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
			this.ClientSize = new System.Drawing.Size(328, 336);
			this.Controls.Add(this.cmdClose);
			this.Controls.Add(this.picPhoto);
			this.Controls.Add(this.label5);
			this.Controls.Add(this.txtNickname);
			this.Controls.Add(this.txtDescription);
			this.Controls.Add(this.txtBirthday);
			this.Controls.Add(this.txtFullname);
			this.Controls.Add(this.label4);
			this.Controls.Add(this.label3);
			this.Controls.Add(this.label2);
			this.Controls.Add(this.label1);
			this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedSingle;
			this.MaximizeBox = false;
			this.Name = "frmVcard";
			this.Text = "frmVcard";
			this.ResumeLayout(false);

		}
		#endregion


		private void VcardResult(object sender, IQ iq, object data)
		{
            if (InvokeRequired)
            {
                // Windows Forms are not Thread Safe, we need to invoke this :(
                // We're not in the UI thread, so we need to call BeginInvoke				
                BeginInvoke(new IqCB(VcardResult), new object[] { sender, iq, data });
                return;
            }
			if (iq.Type == IqType.result)
			{
				Vcard vcard = iq.Vcard;
				if (vcard!=null)
				{
					txtFullname.Text	= vcard.Fullname;
					txtNickname.Text	= vcard.Nickname;
					txtBirthday.Text	= vcard.Birthday.ToString();
					txtDescription.Text = vcard.Description;
					Photo photo = vcard.Photo;
					if (photo != null)
						picPhoto.Image		= vcard.Photo.Image;
				}
                
                
			}
		}

		private void cmdClose_Click(object sender, System.EventArgs e)
		{
			this.Close();
		}
	}
}
